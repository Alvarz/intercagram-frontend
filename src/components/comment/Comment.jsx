import React from 'react'
import './css/Comment.scss'
import CommentCtrl from './js/CommentCtrl'
import { Link } from 'react-router-dom'

export default class Comment extends CommentCtrl {
  constructor (props) {
    super(props)

    this.comment = this.props.comment
  }

  render () {
    return (
      <div className='row'>
        <div className='comment-thumbnail-container' >
          <Link className='' to={`/profile/${this.comment.user}`} >
            <img className='rounded-circle thumb-comment img-thumbnail' src='https://image.shutterstock.com/image-vector/female-profile-picture-placeholder-vector-260nw-450966937.jpg' alt='alt' />
          </Link>
        </div>
        <div className='comment-textbox col'>
          <div className='row'>
            <div className='col-12 align-left'>
              <span>{this.comment.user.name}</span>
            </div>
            <div className='col-12 align-left'>
              {this.comment.description}
            </div>
          </div>
        </div>
      </div>
    )
  }
}
