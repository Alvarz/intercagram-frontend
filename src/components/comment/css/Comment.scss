.navbar-nav-custom{
  width:100%;
  list-style:none;
  flex-wrap: nowrap; 
}

.thumb-comment{
  width:60px;
  height:60px;
  float:left;
}

.comment-thumbnail-container{
  width:60px !important;
}

.comment-textbox{
  float:left;
  width:calc(60px - 100%) !important
}

