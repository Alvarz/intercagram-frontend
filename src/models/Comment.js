
import Requester from '../utils/request'
import to from '../utils/to'

export default class Comment {
  /*
   * class constructor
   */
  constructor () {
    this.requester = new Requester()
    this.baseUrl = `${process.env.REACT_APP_BACKEND_API}/comments`
  }

  /*
   * get the comment of given pic id
   * @param {number} id
   * @return {promise}
   * */
  async get (id) {
    const [ error, response ] = await to(this.requester.get(`${this.baseUrl}/${id}`))
    if (error) return error
    return response.data
  }

  /*
   * fetch the comment of given pic id
   * @param {number} page
   * @return {promise}
   * */
  async fetch (page = 1) {
    const [ error, response ] = await to(this.requester.get(`${this.baseUrl}`))
    if (error) return error
    return response.data
  }

  /*
   * post a new comment
   * @param {object} comment
   * @return {promise}
   * */
  async post (comment) {
    const [ error, response ] = await to(this.requester.post(`${this.baseUrl}`, comment))
    if (error) return error
    return response.data
  }

  /*
   * remove a comment
   * @param {number} id
   * @return {promise}
   * */
  async remove (id) {
    const [ error, response ] = await to(this.requester.remove(`${this.baseUrl}/uncomment/${id}`))
    if (error) return error
    return response.data
  }
}
