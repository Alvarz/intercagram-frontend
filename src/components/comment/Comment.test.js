import React from 'react'
import Comment from './Comment'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'

let initialState = {

}

let comment = {
  user: {
    picProfile: 'http://www.pic.com'
  }
}

it('renders without crashing', () => {
  let mockStore = configureStore()

  // Store
  let store = mockStore(initialState)

  // Wrappeamos el componente
  let wrapper = shallow(<Comment store={store} comment={comment} />, { initialState })

  // Validamos que renderice.
  expect(wrapper).toMatchSnapshot()
})
