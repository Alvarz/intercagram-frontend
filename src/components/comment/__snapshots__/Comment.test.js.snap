// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comment
    comment={
      Object {
        "user": Object {
          "picProfile": "http://www.pic.com",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="comment-thumbnail-container"
        >
          <Link
            className=""
            replace={false}
            to="/profile/[object Object]"
          >
            <img
              alt="alt"
              className="rounded-circle thumb-comment img-thumbnail"
              src="http://www.pic.com"
            />
          </Link>
        </div>,
        <div
          className="comment-textbox col"
        >
          <div
            className="row"
          >
            <div
              className="col-12 align-left"
            >
              <span />
            </div>
            <div
              className="col-12 align-left"
            />
          </div>
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className=""
            replace={false}
            to="/profile/[object Object]"
          >
            <img
              alt="alt"
              className="rounded-circle thumb-comment img-thumbnail"
              src="http://www.pic.com"
            />
          </Link>,
          "className": "comment-thumbnail-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="alt"
              className="rounded-circle thumb-comment img-thumbnail"
              src="http://www.pic.com"
            />,
            "className": "",
            "replace": false,
            "to": "/profile/[object Object]",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "alt",
              "className": "rounded-circle thumb-comment img-thumbnail",
              "src": "http://www.pic.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-12 align-left"
            >
              <span />
            </div>
            <div
              className="col-12 align-left"
            />
          </div>,
          "className": "comment-textbox col",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-12 align-left"
              >
                <span />
              </div>,
              <div
                className="col-12 align-left"
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span />,
                "className": "col-12 align-left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "col-12 align-left",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comment-thumbnail-container"
          >
            <Link
              className=""
              replace={false}
              to="/profile/[object Object]"
            >
              <img
                alt="alt"
                className="rounded-circle thumb-comment img-thumbnail"
                src="http://www.pic.com"
              />
            </Link>
          </div>,
          <div
            className="comment-textbox col"
          >
            <div
              className="row"
            >
              <div
                className="col-12 align-left"
              >
                <span />
              </div>
              <div
                className="col-12 align-left"
              />
            </div>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className=""
              replace={false}
              to="/profile/[object Object]"
            >
              <img
                alt="alt"
                className="rounded-circle thumb-comment img-thumbnail"
                src="http://www.pic.com"
              />
            </Link>,
            "className": "comment-thumbnail-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="alt"
                className="rounded-circle thumb-comment img-thumbnail"
                src="http://www.pic.com"
              />,
              "className": "",
              "replace": false,
              "to": "/profile/[object Object]",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "alt",
                "className": "rounded-circle thumb-comment img-thumbnail",
                "src": "http://www.pic.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col-12 align-left"
              >
                <span />
              </div>
              <div
                className="col-12 align-left"
              />
            </div>,
            "className": "comment-textbox col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-12 align-left"
                >
                  <span />
                </div>,
                <div
                  className="col-12 align-left"
                />,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span />,
                  "className": "col-12 align-left",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "col-12 align-left",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "initialState": Object {},
  },
}
`;
